app/src/main/java/com/example/travelapp/fragments/ComposeFragment.java

```diff
@@ -159,11 +164,16 @@ public void run() {
 
     }
 
-    private Details saveDetails(ParseUser currentUser, Itinerary itinerary) throws InterruptedException {
+    private Details saveDetails(ParseUser currentUser, Itinerary itinerary, boolean visitAll) throws InterruptedException {
         // create graph and call algorithm to determine path
-        Graph graph = createGraph(ids, itinerary);
+        Graph graph = createGraph(ids, itinerary, visitAll);
         // list that stores the indices of locations in order of traversal from locations list
-        List<Integer> listNodesInt = getShortestPath(graph, firstPreference, secondPreference);
+        List<Integer> listNodesInt;
+        if (visitAll) {
+            listNodesInt = findShortestPathAllVertices(graph);
+        } else {
+            listNodesInt = getShortestPath(graph, firstPreference, secondPreference);
+        }
```

Comment:
this is just a personal preference, but with situations like this, i would tend towards doing something like 
`List<Integer> listNodesInt = visitAll ? findShortestPathAllVertices(graph) : getShortestPath(graph, firstPreference, secondPreference);`
it does the exact same thing but is cleaner imo